x-pack/legacy/plugins/apm/public/components/app/ServiceMap/index.tsx

```diff
@@ -37,19 +37,36 @@ ${theme.euiColorLightShade}`,
 
 export function ServiceMap({ serviceName }: ServiceMapProps) {
   const {
-    urlParams: { start, end }
+    urlParams: { start, end, environment },
+    uiFilters
   } = useUrlParams();
 
+  const uiFiltersOmitEnv = useMemo(
+    () => ({
+      ...uiFilters,
+      environment: undefined
+    }),
+    [uiFilters]
+  );
+
   const { data } = useFetcher(
     callApmApi => {
       if (start && end) {
         return callApmApi({
           pathname: '/api/apm/service-map',
-          params: { query: { start, end } }
+          params: {
+            query: {
+              start,
+              end,
+              environment,
```

Comment:
I think my previous comment was lost in a rebase: 
Is it not possible to extract `environment` from `uiFilters` on the backend?